test/framework/src/main/java/org/elasticsearch/test/rest/yaml/Features.java

```diff
@@ -53,11 +56,23 @@ private Features() {
      * Tells whether all the features provided as argument are supported
      */
     public static boolean areAllSupported(List<String> features) {
-        for (String feature : features) {
-            if (!SUPPORTED.contains(feature)) {
-                return false;
+        try {
+            for (String feature : features) {
+                if (feature.equals("xpack")) {
+                    if (false == ESRestTestCase.hasXPack()) {
+                        return false;
+                    }
+                } else if (feature.equals("no_xpack")) {
+                    if (ESRestTestCase.hasXPack()) {
+                        return false;
+                    }
+                } else if (false == SUPPORTED.contains(feature)) {
```

Comment:
any reason we just dont `return SUPPORTED.contains(feature)` for the last if branch?